### PR TITLE
security/polynomial-redos

### PR DESCRIPTION
**Pull request by hubertkozik**
Fixing possible polynomial regular expressions.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207191482240716

### CODE REVIEW

**Comment by hubertkozik** [comment]
I am unsure how to test if this fix will fix the issues listed by the customer. Our CodeQL analysis didn't catch this issues.

File: ts/Extensions/DownloadURL.ts [hunk 1]
```diff
@@ -142,6 +142,10 @@ function downloadURL(
 
     dataURL = '' + dataURL;
 
+    if (nav.userAgent.length > 1000) {
```
Comment (goransle) [inline ts/Extensions/DownloadURL.ts:145]: Not that big of an issue, but should we have `1000` as a static constant somewher? 🤔

**Review by goransle — APPROVED** [review]

File: ts/Extensions/DownloadURL.ts (hunk 1 above)
Comment (hubertkozik) [reply ts/Extensions/DownloadURL.ts:145]: Hmm, that's a few places between different files, maybe someone else has that suggestion?

File: ts/Extensions/OfflineExporting/OfflineExporting.ts [hunk 2]
```diff
@@ -473,51 +473,58 @@ namespace OfflineExporting {
...
+                            /^<svg[^>]*width\s*=\s{,1000}\"?(\d+)\"?[^>]*>/
+                        ),
+                        matchedImageHeight = svg.match(
+                            /^<svg[^>]*height\s*=\s{,1000}\"?(\d+)\"?[^>]*>/
+                        );
+
+                    if (ctx && matchedImageWidth && matchedImageHeight) {
+                        const imageWidth = +matchedImageWidth[1] * scale,
+                            imageHeight = +matchedImageHeight[1] * scale,
+                            downloadWithCanVG = function (): void {
```
Comment (bre1470) [inline ts/Extensions/OfflineExporting/OfflineExporting.ts:493]: Optimization idea: Make it an arrow function.

File: ts/Extensions/DownloadURL.ts [hunk 3]
```diff
@@ -25,6 +25,8 @@ const {
     win: { document: doc }
 } = H;
 
+import { RegexLimits } from './RegexLimits.js';
```
Comment (bre1470) [inline ts/Extensions/DownloadURL.ts:28]: Make use of default export.

File: ts/Extensions/Exporting/Exporting.ts [hunk 4]
```diff
@@ -50,6 +50,7 @@ const {
 } = G;
 import HU from '../../Core/HttpUtilities.js';
 import { Palette } from '../../Core/Color/Palettes.js';
+import { RegexLimits } from '../RegexLimits.js';
```
Comment (bre1470) [inline ts/Extensions/Exporting/Exporting.ts:52]: Make use of default export.

File: ts/Extensions/RegexLimits.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Limits for regex check to prevent overload of regex engine.
+ */
+export const enum RegexLimits {
+    shortLimit = 1000,
+    svgLimit = 100000000
+}
```
Comment (bre1470) [inline ts/Extensions/RegexLimits.ts:7]: Make use of default export.

**Review by bre1470 — APPROVED** [review]
One thing to do, otherwise good. 👍

File: ts/Extensions/RegexLimits.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Limits for regex check to prevent overload of regex engine.
+ */
+const enum RegexLimits {
+    shortLimit = 1000,
+    svgLimit = 100000000
```
Comment (KacperMadej) [inline ts/Extensions/RegexLimits.ts:6]: `1e8` or `100_000_000` seems more readable

**Review by KacperMadej — APPROVED** [review]

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**